File: 151-reverse-words-in-a-string/reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    let res = "";
    for (let end = s.length - 1; end >= 0; end--) {
      if (s[end] !== ' ') {
        let start = end;
        while (s[start] !== ' ' && start >= 0) {
          start--;
        }
        let word = s.slice(start + 1, end + 1);
        if (res !== '') res += (' ');
        res += word;
        end = start;
      }
    }
    return res;
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        words = []
        word = []
        for char in s:
            # if char is ' ' - if word, add word to words
            if char == ' ':
                if word:
                    words.append("".join(word))
                    word = []
            # else add char to word
            else:
                word.append(char)
        
        # if word, add word to words
        if word:
            words.append("".join(word))

        print('words-', words)
        
        return " ".join(words[::-1])